### AI-04 Practice 3  
#### Sample notebook of Poisson regression
ポアソン回帰分析の手順例  

#### Import libraries

In [1]:
import pandas as pd
import statsmodels.api as sm

#### Parameters  

In [2]:
csv_in = 'medical_accident.csv'

#### Read CSV file  

In [3]:
df = pd.read_csv(csv_in, delimiter=',', skiprows=0, header=0)
print(df.shape)
print(df.info())
display(df.head())

(100, 5)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 5 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   env1              100 non-null    float64
 1   env2              100 non-null    float64
 2   env3              100 non-null    float64
 3   env4              100 non-null    float64
 4   medical_accident  100 non-null    int64  
dtypes: float64(4), int64(1)
memory usage: 4.0 KB
None


env1  env2  env3  env4  medical_accident
0   5.3   0.3   4.9   6.6                 1
1   4.5   4.2   4.5   1.4                 0
2   5.1   3.1   9.5   6.6                 1
3   4.0   1.9  -3.2   6.9                 1
4   3.9   3.5   0.1   6.1                 1

#### Separate explanatory variables and objective variable  
説明変数と目的変数を分ける    

In [4]:
X = df.loc[:, 'env1':'env4']
y = df['medical_accident']
X_c = sm.add_constant(X)

#### Poisson regression  

In [5]:
model = sm.GLM(y, X_c, family=sm.families.Poisson())
results = model.fit()
print(results.summary())

                 Generalized Linear Model Regression Results                  
Dep. Variable:       medical_accident   No. Observations:                  100
Model:                            GLM   Df Residuals:                       95
Model Family:                 Poisson   Df Model:                            4
Link Function:                    log   Scale:                          1.0000
Method:                          IRLS   Log-Likelihood:                -73.069
Date:                Tue, 28 Apr 2020   Deviance:                       33.524
Time:                        22:14:59   Pearson chi2:                     26.8
No. Iterations:                     5                                         
Covariance Type:            nonrobust                                         
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const         -3.4011      0.952     -3.572      0.0

#### Regression coefficients  

In [6]:
print(results.params)

const   -3.401070
env1     0.238637
env2     0.209806
env3     0.009130
env4     0.194961
dtype: float64


**Ans.
$\ln(\rm{medical\_accident}) = (0.24) * \rm{env1} + (0.21) * \rm{env2} + (0.01) * \rm{env3} + (0.19) * \rm{env4} -0.34$**